lib/chore/queues/sqs/publisher.rb

```diff
@@ -15,9 +15,27 @@ def initialize(opts={})
         end
 
         # Takes a given Chore::Job instance +job+, and publishes it by looking up the +queue_name+.
-        def publish(queue_name,job)
+        def publish(queue_name, job, options={})
+          delay_seconds = options[:delay]
+          options = {}
+
+          # SQS only supports a delay range of 0-900 seconds. Make sure we're in the range. We also only put it into the
+          # options Hash if its set to prevent any weird SQS behaviors.
+          # NOTE may be okay pushing the key with a `nil` value...
+          if delay_seconds
+            if delay_seconds < 0
+              delay_seconds = 0
+            elsif delay_seconds > 900
+              Chore.logger.warn("SQS does not support a delay longer than 15 minutes (900 seconds)! " <
+                                "Reducing the delay to the maximum supported.")
+              delay_seconds = 900
```

Comment:
I actually think we want to raise an error here. Any time we do a thing SQS doesn't support, we raise an error from what I recall (either we do, or SQS does because we're not guarding against it)

For example: https://github.com/Tapjoy/chore_internal/blob/master/lib/chore/queues/sqs/consumer.rb#L20